Automattic/wp-calypso
# Reduxify: lib/username

Trying to fix #24128.
This PR is also part of #24004 as it removes `lib/user` from `lib/username`.
- Reduxifies `lib/username`
- Uses a query component to send network requests for username validation
- Decouples `lib/user` from `lib/username` by moving the relevant logic to `me/account/main.jsx`
- Adds unit test cases

**To Test**
- Open http://calypso.localhost:3000/me/account
- Type 3 characters in the **Username** input box and confirm that an error message `Usernames must be at least 4 characters.` appears
- Type random characters with invalid characters (i.e. not lowercase letters and not numbers) in the **Username** input box and confirm that an error message `
Usernames can only contain lowercase letters (a-z) and numbers.` appears
- Type an existing username in the **Username** input box and confirm that an error message `Sorry, that username already exists!` appears
- Type a valid and non-existing username in the **Username** input box and confirm that a message `<username> is a valid username.` appears
- Click the **Save Username** button at the bottom of the form and confirm that the username is changed after page reload
